scripts/node/refreshDatabase.js

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env node
+
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+const Script = require('./Script');
+
+class RefreshDatabaseScript extends Script {
+  config = {
+    command: '* <dumpPath>',
+    options: {
+      migrate: {
+        alias: 'm',
+        type: 'boolean',
+        description: 'Run migrations after DB has been refreshed',
+      },
+    },
+    version: '1.0.0',
+  };
+
+  runCommands() {
+    this.refreshDatabase();
+    if (this.args.migrate) {
+      this.runMigrations();
+    }
+  }
+
+  refreshDatabase() {
+    this.logInfo('Refreshing the tupaia database...');
+
+    this.verifyPsql();
+    this.exec('psql -U postgres -c "DROP DATABASE IF EXISTS tupaia"');
+    this.exec('psql -U postgres -c "CREATE DATABASE tupaia WITH OWNER tupaia"');
+    this.exec('psql -U postgres -d tupaia -c "CREATE EXTENSION postgis"');
+    this.exec('psql -U postgres -c "ALTER USER tupaia WITH SUPERUSER"');
```

Comment:
You've probably thought about it but these are in the opposite order to in the bash script. (I don't normally run either of these 😬)

Reply:
The first one will be required only in the initial setup of the database, but doesn't harm running in each consequent case. So it's fine you don't run it 🙂

The order here doesn't matter from a functionality perspective, but I believe that the new order makes the command flow more readable.

I guess you do have to have to run `ALTER USER...`, since if not you'll get a bunch of errors? Unless if you have made `tupaia` a SUPERUSER in your local db 😉

Reply:
Yeah thanks, that makes sense!

And yeah, I never knew that I should make tupaia not a super user 😁...

Reply:
Well, it may not create an issue ever, but it's good to replicate the actual DB as closely as possible. Eg it may be that a specific command you run succeeds in your local DB, but not in our CI/CD pipeline, because it depends on admin rights.

Reply:
And those are usually the frustrating category of issues to solve 😆

Reply:
Yeah thanks, I never knew!

That's the hidden (or not so hidden perhaps) advantage of these utility scripts. Making sure that everyone's process is consistent to minimize the 'works on my machine' problem.